src/Control/Middleware/TrustedProxyMiddleware.php

```diff
@@ -220,14 +223,14 @@ protected function getIPFromHeaderValue($headerValue)
 
         // Prioritise filters
         $filters = [
-            FILTER_FLAG_NO_PRIV_RANGE | FILTER_FLAG_NO_RES_RANGE,
-            FILTER_FLAG_NO_PRIV_RANGE,
-            null
+            Ip::ALL_ONLY_PUBLIC,
+            Ip::ALL_NO_PRIVATE,
+            Ip::ALL
         ];
         foreach ($filters as $filter) {
             // Find best IP
             foreach ($ips as $ip) {
-                if (filter_var($ip, FILTER_VALIDATE_IP, $filter ?? 0)) {
+                if (ConstraintValidator::validate($ip, [new Ip(version: $filter), new NotBlank()])->isValid()) {
                     return $ip;
                 }
             }
```

Comment:
This ends up being the exact same thing under the hood - though frankly Symfony have a better track record of keeping up with RFCs than PHP internal functions do so this seems better long-term.